godtools/App/Features/ToolsFilter/Presentation/ToolFilterLanguageSelectionView/ToolFilterLanguageSelectionViewModel.swift

```diff
@@ -50,7 +48,21 @@ class ToolFilterLanguageSelectionViewModel: ObservableObject {
         $appLanguage.eraseToAnyPublisher()
             .flatMap { appLanguage in
                 
-                return viewToolFilterLanguagesUseCase.viewPublisher(filteredByCategoryId: selectedCategory.id, translatedInAppLanguage: appLanguage)
+                return getUserToolFiltersUseCase.getUserToolFiltersPublisher(translatedInAppLanguage: appLanguage)
+                    .eraseToAnyPublisher()
+            }
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] userFilters in
+            
+                self?.selectedLanguage = userFilters.languageFilter
+                self?.selectedCategory = userFilters.categoryFilter
+            }
+            .store(in: &cancellables)
+        
+        $appLanguage.eraseToAnyPublisher()
```

Comment:
Hey @rachaelblue it looks like here you will want to react to selected category changes as well to push into the use case.

Reply:
Hey @levieggertcru, the category won't get changed while you're on the language selection page.  You'd have to back out of the languages, change the category, and then reopen it.  So I don't think the category publisher would ever fire.  Same with the category page and language selection.